l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.py

```diff
@@ -0,0 +1,388 @@
+            'issued_by': issued_by,
+            'company_vat': company_vat,
+            'comp_name': comp_name,
+            'street': street,
+            'zip': _zip,
+            'city': city,
+            'country': country,
+            'email': email,
+            'phone': phone,
+            'SenderId': SenderId,
+            'period': self.env.context['year'],
+            'comments': data['comments'] or ''
+        }
+
+        data_file = """<?xml version="1.0" encoding="ISO-8859-1"?>
+<ns2:ClientListingConsignment xmlns="http://www.minfin.fgov.be/InputCommon"
+ xmlns:ns2="http://www.minfin.fgov.be/ClientListingConsignment"
+ ClientListingsNbr="1">
+    <ns2:Representative>
+        <RepresentativeID identificationType="NVAT"
+         issuedBy="%(issued_by)s">%(SenderId)s</RepresentativeID>
+        <Name>%(comp_name)s</Name>
+        <Street>%(street)s</Street>
+        <PostCode>%(zip)s</PostCode>
+        <City>%(city)s</City>"""
+        if annual_listing_data['country']:
+            data_file += "\n\t\t<CountryCode>%(country)s</CountryCode>"
+        data_file += """
+        <EmailAddress>%(email)s</EmailAddress>
+        <Phone>%(phone)s</Phone>
+    </ns2:Representative>"""
+        data_file = data_file % annual_listing_data
+
+        data_comp = """
+        <ns2:Declarant>
+            <VATNumber>%(SenderId)s</VATNumber>
+            <Name>%(comp_name)s</Name>
+            <Street>%(street)s</Street>
+            <PostCode>%(zip)s</PostCode>
+            <City>%(city)s</City>
+            <CountryCode>%(country)s</CountryCode>
+            <EmailAddress>%(email)s</EmailAddress>
+            <Phone>%(phone)s</Phone>
+        </ns2:Declarant>
+        <ns2:Period>%(period)s</ns2:Period>
+        """ % annual_listing_data
+
+        # Turnover and Farmer tags are not included
+        client_datas = self._get_datas()
+        if not client_datas:
+            raise UserError(_('No data available for the client.'))
+        data_client_info = ''
+        for amount_data in client_datas:
+            data_client_info += """
+    <ns2:Client SequenceNumber="%(seq)s">
+        <ns2:CompanyVATNumber issuedBy="BE">%(only_vat)s</ns2:CompanyVATNumber>
+        <ns2:TurnOver>%(turnover)s</ns2:TurnOver>
+        <ns2:VATAmount>%(vat_amount)s</ns2:VATAmount>
+    </ns2:Client>""" % amount_data
+
+        amount_data_begin = client_datas[-1]
+        amount_data_begin.update({'dnum': dnum})
+        data_begin = """
+    <ns2:ClientListing SequenceNumber="1" ClientsNbr="%(seq)s"
+     DeclarantReference="%(dnum)s"
+        TurnOverSum="%(sum_turnover)s" VATAmountSum="%(sum_tax)s">
+""" % amount_data_begin
+
+        data_end = """
+
+        <ns2:Comment>%(comments)s</ns2:Comment>
+    </ns2:ClientListing>
+</ns2:ClientListingConsignment>
+""" % annual_listing_data
+
+        data_file += data_begin + data_comp + data_client_info + data_end
+        file_save = base64.encodestring(data_file.encode('utf8'))
```

Comment:
@andreparames here encode('ISO-8859-1') instead of utf8 as that's what indicated in the XML declaration above.

Reply:
While this is clearly problematic, it was already outputting in UTF-8, so isn't it risky to change now? Should we change the declaration instead?

Reply:
Ok, let's try with changing the declaration